### DISCUSSION TITLE
How to trigger workflow by push in other repository?

### DISCUSSION THREAD

**Discussion by micash545** [woodpecker-ci/woodpecker#2838] [Ask a Question]
If there is a push to repo 1, the workflow in repo 2 starts.

**Comment by qwerty287** [comment]
Did you see https://woodpecker-ci.org/plugins/Trigger?

**Reply by micash545** [reply]
Yes, but I mean that repo 1 is a mirror, and I can only add a webhook to it.
Gitea BTW.

**Comment by lafriks** [comment]
I don't think that's possible. The only option I can think of would be to use an external service that would return pipeline definition for that repo, I don't know if such option was implemented tho

**Reply by qwerty287** [reply]
Yes, using external config services this could be possible. Docs are available at https://woodpecker-ci.org/docs/next/administration/external-configuration-api

**Reply by micash545** [reply]
So Woodpecker is requesting [Example request made by Woodpecker](https://woodpecker-ci.org/docs/administration/external-configuration-api#example-request-made-by-woodpecker) from $WOODPECKER_CONFIG_SERVICE_ENDPOINT. Can I respond to this with a modified query with a changed repo.name from repo 1 and other values to trigger the pipeline on repo 2?

**Reply by qwerty287** [reply]
No, response looks different: https://woodpecker-ci.org/docs/administration/external-configuration-api#example-response-structure

It actually just returns the content of the yaml files.

**Comment by micash545** [comment]
What about replacing webhooks? I know it may not work as I expected, but I think it's worth a try. 
Where can i generate secret.
![image](https://github.com/woodpecker-ci/woodpecker/assets/70768913/e0952a63-1f32-4630-9d4f-12d5146f072c)

**Reply by qwerty287** [reply]
You can't, this secret is generated internally on webhook creation. Copying isn't possible as well as it's the signed repo name.

**Comment by qwerty287** [comment]
I'm going to close this. The only way how you can do this is a config-service and we don't plan to support an extra way of pipeline definitions currently. Maybe organization/global pipelines could fix this as well when we decide to implement them.
